### PR TITLE
Bug/fix signout

### LINKED ISSUE #142: Sign out button does not remove credentials
**Describe the bug**
If a user is logged in and then clicks the sign out button they are redirected to the sign in page.
However, if the user hits the back button or enters the link for the transaction page they will still be logged in.

**To Reproduce**
1. Login
2. Log out
3. Hit the back button, or go to any other page.
4. You will be logged in as the last user

**Severity**
Mark with an x.
[]: Minor effect eg. graphical
[]: Functional error eg. App does not function correctly
[x]: Severe eg. Crashing

**Reproducibility**
Mark with an x.
[x]: Consistent
[]: Occasional
[]: Cannot reproduce at the moment eg. unsure

**Expected behavior**
The user should not be logged in, as the user has opted to log out. Credentials should be thrown away.

**Conditions**
List the setup you used to create this bug, eg OS, hardware, browser
Ubuntu, Firefox

**Additional context**
Session data seems to be stored in the back end, an API call may be necessary to let the back end know to throw away the session data.

### PR DESCRIPTION
**Pull request by GrunWong**
Closes #142 

**Description**

- Added logout API to backend + cookie and session expiration upon API call
- Changed frontend "Sign Out" button to call backend API + sets user context accordingly

**Testing**

- Tested backend API initially with Postman, tested frontend by using web UI (logging in and out).
- Cookies were tested by using alternative tabs to access other pages (besides login page) after a signout.
- Sessions were tested by using multiple browsers (Mozilla Firefox and Chromium) with different accounts logging in and logging out individually

**Checklist**

- [X] I have tested my change
- [X] Code builds successfully and all tests pass
- [X] Docs have been added/updated if needed

### CODE REVIEW

File: split-webapp/split/src/pages/Home.js [hunk 1]
```diff
@@ -4,8 +4,34 @@ import Transactions from "./Transactions";
...
+const logout = (setUsername) => {
+  fetch("/api/account/logout", {
+    method: "POST",
+    body: "",
+    headers: {
+      "Content-Type": "application/json"
+    }
+  })
+  .then(res => {
+    return res.json()
```
Comment (ErnWong) [inline split-webapp/split/src/pages/Home.js]: nitpick:

image

**Review by ErnWong — APPROVED** [review]
Looks good to me!

File: split-webapp/split/src/pages/Home.js (hunk 1 above)
Comment (GrunWong) [reply split-webapp/split/src/pages/Home.js]: Whoops! How embarrassing. Fixed now.

**Comment by JamesUOA** [comment]
LGTM 👍

**Review by JamesUOA — APPROVED** [review]

**Pull request merged**